guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,872 @@
+
+- The use of a custom callback
+- Using a [SideCarEvaluator](https://www.tensorflow.org/api_docs/python/tf/keras/utils/SidecarEvaluator)
+
+Let's use a custom callback to achieve TPU compatibility in this guide:
+"""
+
+
+class EvaluateCOCOMetricsCallback(keras.callbacks.Callback):
+    def __init__(self, data):
+        super().__init__()
+        self.data = data
+        self.metrics = keras_cv.metrics.BoxCOCOMetrics(
+            bounding_box_format="xywh",
+            # passing 1e9 ensures we never evaluate until
+            # `metrics.result(force=True)` is
+            # called.
+            evaluate_freq=1e9,
+        )
+
+    def on_epoch_end(self, epoch, logs):
+        self.metrics.reset_state()
+        for batch in tqdm.tqdm(self.data):
+            images, y_true = batch[0], batch[1]
+            y_pred = self.model.predict(images, verbose=0)
+            self.metrics.update_state(y_true, y_pred)
+
+        metrics = self.metrics.result(force=True)
+        logs.update(metrics)
+        return logs
+
+
+"""
+Our data pipeline is now complete!
+We can now move on to model creation and training.
+
+## Model creation
+
+Next, let's use the KerasCV API to construct an untrained RetinaNet model.
+In this tutorial we using a pretrained ResNet50 backbone from the imagenet
+dataset.
+
+KerasCV makes it easy to construct a `RetinaNet` with any of the KerasCV
+backbones.  Simply use one of the presets for the architecture you'd like!
+
+For example:
+"""
+
+model = keras_cv.models.RetinaNet.from_preset(
+    "resnet50_imagenet",
+    num_classes=len(class_mapping),
+    # For more info on supported bounding box formats, visit
+    # https://keras.io/api/keras_cv/bounding_box/
+    bounding_box_format="xywh",
+)
+
+"""
+That is all it takes to construct a KerasCV RetinaNet.  The RetinaNet accepts
+tuples of dense image Tensors and bounding box dictionaries to `fit()` and
+`train_on_batch()`
+
+This matches what we have constructed in our input pipeline above.
+"""
+
+
+"""
+## Training our model
+
+All that is left to do is train our model.  KerasCV object detection models
+follow the standard Keras workflow, leveraging `compile()` and `fit()`.
+
+Let's compile our model:
+"""
+model.compile(
+    classification_loss="focal",
+    box_loss="smoothl1",
+    optimizer=optimizer,
+    # We will use our custom callback to evaluate COCO metrics
+    metrics=None,
+)
+"""
+If you want to train the fully model, uncomment `.take(20)` from each
+of the following dataset references.
+"""
+model.fit(
+    train_ds.take(20),
+    validation_data=eval_ds.take(20),
+    # Run for 10-35~ epochs to achieve good scores.
+    epochs=1,
+    callbacks=[EvaluateCOCOMetricsCallback(eval_ds.take(20))],
+)
+"""
+
+## Inference and plotting results
+
+KerasCV makes object detection inference simple.  `model.predict(images)`
+returns a RaggedTensor of bounding boxes.  By default, `RetinaNet.predict()`
```

Comment:
can you be more explicit about the retuned format here ? There are classes too aren't there ?